### DISCUSSION TITLE
Unclear if tycho-surefire now provides correct JaCoCo coverage eclipse-testplugin (id=397015)

### DISCUSSION THREAD

**Discussion by CasperNLD** [eclipse-tycho/tycho#1546] [Q&A]
HI,
 I was wondering about a requested feature some time ago.
https://bugs.eclipse.org/bugs/show_bug.cgi?id=397015

In my case we have a eclipse-plugin which is tested with a eclipse-test plugin. So the test coverage which is generated by the test plugin is in fact the results of the eclipse plugin. 
Since the migration to SonarQube 9.x from jacoco.exec to jacoco.xml, the workaround  with groovy altering runtime the path is not working anymore.

We are still unable to check with Tycho 3.0, but I was hoping that Tycho now has a neat way to configure this.
Can someone tell if in the mean time there is an better way? May be perhaps add this howto in the online tycho documentation?

Thanks in advance for the replies.

**Comment by laeubi** [comment]
Eclipse has the notation of "test" folders so this could be used to map to maven test_sources. Can you
provide an [integration-test](https://github.com/eclipse/tycho/wiki#providing-an-integration-test) to demonstrate the issue?

**Comment by CasperNLD** [comment]
Hi, 
I've been making a clean (distilled) test-case for the integration-test since your reply and as you have stated moving to 'test'-directories. seems to work now. I have to discuss with my sonar-group if we can implement it on a larger scale to see if we run into principal considerations.
Enclosed you find a simplified test-case, I did not have had time to implement a full integration test and perhaps it will have a use in the near future.
[eclipse_pluginblank.zip](https://github.com/eclipse-tycho/tycho/files/9827860/eclipse_pluginblank.zip)

**Reply by CasperNLD** [reply]
If you have any follow-ups regarding this issue, you can use this issue. I have to check internally which can take some time. So this item can be closed to avoid any longstanding open issues.
